src/qir2qasm_trans/qir_trans/translator.py

```diff
@@ -0,0 +1,343 @@
+
+        statements = (
+            include_statements + var_decl_statements + func_decl_statements + main_statements
+        )
+        return ast.Program(version="3.0", statements=statements)
+
+    def build_var_declarations(self) -> List[ast.Statement]:
+        building_task: List[Tuple[DeclBuilder, str, int]] = []
+        for name, struct_info in self.symbols.structures.items():
+            builder = struct_info.decl_builder
+            building_task.append((builder, f"{name}s", self.symbols.structures_num[name]))
+            building_task.append((builder, f"{name}s_tmp", self.symbols.structures_tmp_num[name]))
+
+        for name, type_ast in self.symbols.classical_tmp.items():
+            builder = ClassicalDeclarationBuilder(type_ast)
+            building_task.append((builder, f"{name}_tmp", self.symbols.classical_tmp_num[name]))
+
+        statements = []
+        for builder, name, size in building_task:
+            if size > 0:
+                statements.append(builder.building(name, size))
+
+        for io_key in ["input", "output"]:
+            for type_ast, ident_ast in self.symbols.io_variables[io_key]:
+                statements.append(
+                    ast.IODeclaration(
+                        io_identifier=ast.IOKeyword[io_key], type=type_ast, identifier=ident_ast
+                    )
+                )
+
+        return statements
+
+    def build_func_declarations(self) -> List[ast.Statement]:
+        module = self.module
+
+        statements_declaration = []
+        statements_definition = []
+
+        for name, struct_info in self.profile.structures.items():
+            self.symbols.register_structure(name, struct_info)  # Register Structure in SymbolTable
+
+        for name, inst_info in self.profile.classical_instruction.items():
+            self.symbols.instructions[name] = inst_info  # Register Instruction in SymbolTable
+
+        for func in module.functions:
+            if func.is_declaration:
+                func_info = self.profile.standard_functions.get(func.name)
+
+                if func_info:
+                    # If func is defined in QIR Profile
+                    self.symbols.functions[func.name] = (
+                        func_info  # Register Function in SymbolTable
+                    )
+                    def_statement = func_info.def_statement
+                    if def_statement:
+                        statements_definition.append(def_statement)
+                else:
+                    # If func is not defined in QIR Profile
+                    func_name = func.name
+                    func_type = func.type.element_type
+
+                    # Build the "defcal" QASM ast node
+                    arguments = []
+                    qubits = []
+                    num_qubits = 0
+
+                    arg_types = [arg_type for arg_type in func_type.elements]
+
+                    # Return Type
+                    ret_type = arg_types[0]
```

Comment:
Does QIR only allow 1 return? What if there are more than 1 return?

Reply:
If there is multiple reutrn object, it's a list. The return object must be the first element